drivers/gles2/shaders/scene.glsl

```diff
@@ -1121,7 +1119,7 @@ LIGHT_SHADER_CODE
 	float NdotL = dot(N, L);
 	float cNdotL = max(NdotL, 0.0); // clamped NdotL
 	float NdotV = dot(N, V);
-	float cNdotV = max(NdotV, 0.0);
+	float cNdotV = max(abs(NdotV), 1e-6);
```

Comment:
Note: The absolute value is not necessary for clamping to a minimum of 1e-6. I'm not saying it should be removed, you might have added it for another reason, but just know it's not necessary for enforcing a minimum of 1e-6.

Reply:
abs is not needed but it might produce results that look a little better (still having some specular variations at very steep angles instead of clipping the value)
Though I'm also interested in why this has been added 😄

Reply:
The major reason for the minimum is that I am not an expert on these shaders and Filament does something similar. Filament, which is mentioned in the comments inside Godot's shader code, has this two lines:
```
float NoV = abs(dot(n, v)) + 1e-5;
float NoL = clamp(dot(n, l), 0.0, 1.0);
```
So I just went with the minimum.